log-collector/src/main/scala/io/phdata/pulse/logcollector/KuduMetricStream.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2018 phData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.phdata.pulse.logcollector
+
+import java.util.{ ArrayList, Collections }
+
+import io.phdata.pulse.common.domain.TimeseriesEvent
+import org.apache.kudu.client.SessionConfiguration.FlushMode
+import org.apache.kudu.client.{ CreateTableOptions, KuduClient, KuduException, KuduTable }
+import org.apache.kudu.{ ColumnSchema, Schema, Type }
+
+import scala.collection.concurrent
+
+object TimeseriesEventColumns {
+  val TIMESTAMP = "timestamp"
+  val METRIC    = "metric"
+  val VALUE     = "value"
+}
+
+/**
+ * A Kudu Metric stream batches data by time and record count into Kudu tables.
+ * @param client
+ */
+class KuduMetricStream(client: KuduClient) extends Stream[TimeseriesEvent] {
+  lazy val session = {
+    val session = client.newSession()
+    // Flushing is handled by the stream.
+    session.setFlushMode(FlushMode.MANUAL_FLUSH)
+    session
+  }
+
+  class KuduRowErrorException(message: String) extends Exception(message)
```

Comment:
`case class` ?

Reply:
I'm not using pattern matching on this exception yet, but good idea. Added.